src/tools/compiletest/src/header.rs

```diff
@@ -942,6 +943,7 @@ pub fn make_test_description<R: Read>(
             && config.parse_name_directive(ln, "needs-sanitizer-support");
         ignore |= !has_asan && config.parse_name_directive(ln, "needs-sanitizer-address");
         ignore |= !has_cfi && config.parse_name_directive(ln, "needs-sanitizer-cfi");
+        ignore |= !has_kasan && config.parse_name_directive(ln, "needs-sanitizer-kernel-address");
```

Comment:
Hmm, this directive is difficult because we'll never see that in CI unless we `test --target x86_64-unknown-none`, and we don't ever do that now because most of the other ui and codegen tests assume `std` is available.

But your tests don't really need any of that. I suggest dropping them all the way down to `#![no_core]` and passing explicit `--target` flags within the test, and you can hit all of the enabled targets this way. You can look at `codegen/stack-probes-inline.rs` as a small example of this approach.

Reply:
Done :+1: